File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.py
# def solution(n):
#     if n < 2:
#         return n

#     k = (n + 1) // 2 + 1
#     answer = 0

#     for i in range(1, k):
#         total = 0
#         j = i
        
#         while total < n:
#             total += j
#             j += 1

#         answer = answer + 1 if total == n else answer

#     return answer + 1
# logN * N

def solution(n):
    cnt = 0
    k = int((n // 2) + 1)

    for i in range(1, k):
        for j in range(i + 1, n):
            i += j
            
            if i > n:
                break
            elif i == n:
                cnt += 1
                break
    
    return cnt + 1
    

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
public int solution(int n) {
        int answer = 0;

        for (int i = 1; i <= n; i++) {
            int total = i;

            for (int j = i+1; j <= n; j++) {
                total += j;

                if (total >= n) {
                    if (total == n) {
                        answer++;
                    }
                    break;
                }
            }
        }
        return answer + 1;
    }
}